frontend-react/docs/proposals/0003-uswds-react-components.md

```diff
@@ -0,0 +1,251 @@
+# [0003: Creating New React Components]
+
+* Status: [proposed | rejected | accepted | deprecated | superseded by [0005](0005-example.md)]
+* Authors: @etanb
+* Deciders: TBD
+* Proposal date: 2023-01-24
+* Last updated: [YYYY-MM-DD when the proposal was last updated]
+
+## Problem Statement
+
+- We rely on [USWDS components](https://designsystem.digital.gov/components/overview/) for the basis of much of the code in ReportStream. This creates a rigidity in our codebase since our designs require extensions on their library and contributing to their codebase has proved a cumbersome and untimely process.
+- On top of the base USWDS library, we also use [Trussworks React component library](https://github.com/trussworks/react-uswds). This adds another level of rigidity to the stack since we aren't interacting with the base USWDS library, we're only exposed to whatever `props` trussworks exposes for us.
+- From-the-ground-up customized components while offering complete control, could move us too far off the standardized, accessible components offered by the USWDS + would be quite a lot of work to create and maintain.
+
+## Decision Drivers
+
+- Recent experience adding a simple addition to USWDS showed that we can't rely on them to update their library according to our needs.
```

Comment:
Something related to these points is that Trussworks's MO is to provide very thin, almost logic-less wrappers around the basic USWDS markup; they specifically don't want to deviate too far from the basic markup.  This forces anyone who wants to support more complex UX to either finagle a lot of logic to wrap/consume their components or just create their own custom components.

Reply:
I think I remember reading an issue in which they said this flat out -- will do some digging to find it again if it exists!  Otherwise, they have a [Non-Goals section](https://github.com/trussworks/react-uswds#non-goals) that basically says that the components are purely visual as opposed to having any more functionality 🤔